yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -10,108 +10,39 @@
+from ..converter.converter.converter import Runner as ConvRunner
 
-Geometry Mesh
-    Name MyMesh_YZ
-    X -5.0  5.0    {mesh_nx:d}
-    Y -5.0  5.0    {mesh_ny:d}
-    Z  0.0  30.0   {mesh_nz:d}
 
-Output
-    Filename mesh.bdo
-    Geo ScoringCylinder
-    Quantity Dose
-"""
-input_cfg_templ['geo.dat'] = """
-*---><---><--------><------------------------------------------------>
-    0    0           protons, H2O 30 cm cylinder, r=10, 1 zone
-*---><---><--------><--------><--------><--------><--------><-------->
-  RCC    1       0.0       0.0       0.0       0.0       0.0      30.0
-                10.0
-  RCC    2       0.0       0.0      -5.0       0.0       0.0      35.0
-                15.0
-  RCC    3       0.0       0.0     -10.0       0.0       0.0      40.0
-                20.0
-  END
-  001          +1
-  002          +2     -1
-  003          +3     -2
-  END
-* material codes: 1 - liquid water (ICRU material no 276), 1000 - vacuum, 0 - black body
-    1    2    3
-    1 1000    0
-"""
-
-
-def run_shieldhit(param_dict):
+def run_shieldhit(param_dict, raw_input_dict):
     """Shieldhit runner"""
-    import pymchelper
-
-    input_dict = input_cfg_templ.copy()
-
-    input_dict['beam.dat'] = input_dict['beam.dat'].format(
-        energy=param_dict['energy'],
-        nstat=param_dict['nstat']
-    )
-
-    input_dict['detect.dat'] = input_dict['detect.dat'].format(
-        cyl_nr=param_dict['cyl_nr'],
-        cyl_nz=param_dict['cyl_nz'],
-        mesh_nx=param_dict['mesh_nx'],
-        mesh_ny=param_dict['mesh_ny'],
-        mesh_nz=param_dict['mesh_nz']
-    )
-
     # create temporary directory
     with tempfile.TemporaryDirectory() as tmp_output_path:
 
-        for config_filename in input_dict:
-            abs_input_path = os.path.join(tmp_output_path, config_filename)
-            with open(abs_input_path, 'w') as temp_input_file:
-                temp_input_file.write(input_dict[config_filename])
+        conv_runner = ConvRunner(parser=DummmyParser(),
+                                 input_data=raw_input_dict,
+                                 output_dir=tmp_output_path)
+
+        conv_runner.run_parser()
 
+        print(os.listdir(tmp_output_path))
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
-        isRunOk = runner_obj.run(settings=settings)
-        if not isRunOk:
+        if not runner_obj.run(settings=settings):
```

Comment:
This is a bad idea. Working with debugger is much easier if status code is extracted into separate value, then this value is used in `if` statement